bin/convertToPng

```diff
@@ -42,7 +42,7 @@ mkdir -p "${TEMP_DIR}"
 
 # check if .ico is a sequence
 # pipe into cat so no exit code is given for grep if no matches are found
-IS_ICO_SET="$(identify ${SOURCE} | grep -e "\w\.ico\[0" | cat )"
+IS_ICO_SET="$(identify "${SOURCE}" | grep -e "\w\.ico\[0" | cat )"
```

Comment:
@bacongravy `identify` is part of ImageMagick. If you wish to improve #199, what about having us keeping depending on ImageMagick and fail better if it's absent (detect its absence, offer guidance to install it, etc.). Also,

- You're on your own on this one, I don't have a Mac to test it. Are you confident about it? If not, please ask someone (maybe running a macOS version different from yours) to run it.
- Can you review the [references to imagemagick](https://github.com/jiahaog/nativefier/search?utf8=%E2%9C%93&q=imagemagick&type=) around the codebase and kill those that become unnecessary?
- Please [shellcheck](https://www.shellcheck.net/) your scripts.
    - Super bonus: add shellcheck to our CI run 🙂.

Reply:
I didn't modify the `convertToPng` or `convertToIco` scripts to use `sips`, because they aren't used in the simple path of building on `darwin` for `darwin`.

The `convertToPng` script is used on `darwin` when building for `linux`; the `convertToIco` script is used on `darwin` when building for `win32`. Those scripts still require ImageMagick, as you point out.

I can see a couple of options:

1. Update those scripts to work on `darwin` without requiring ImageMagick
1. Update nativefier to check for ImageMagick and provide a helpful error message when asked to perform a conversion for which it has a missing dependency.

I can understand wanting to keep #199 open until option (1) is implemented, but do you have any objections to merging this PR as-is in the meantime?

It seems like option (2) is a new enhancement request that should be tracked with a new issue.

Reply:
To answer your other points:
* I will have a friend validate these changes on a different Mac before marking this PR as ready for merge.
* I will review the mentions of ImageMagick around the project and update them as appropriate.
* I will shellcheck the scripts.

Reply:
I've run the script through shellcheck (as a `bash` script, not a `sh` script) and updated the PR based on the feedback I got.

Reply:
@bacongravy good point, yes let's make it clear we're using bash in the scripts with a saner hashbang; we were using non-`sh` syntax anyway. Make the scripts start with `#!/usr/bin/env bash` . If unfamiliar, see [this SO page](https://stackoverflow.com/questions/10376206/what-is-the-preferred-bash-shebang) for details.